src/core/ruby_spy.rs

```diff
@@ -123,6 +165,103 @@ impl RubySpy {
             self.process.pid,
         )
     }
+
+    #[cfg(any(windows, target_os = "linux"))]
+    fn get_native_frames(
+        &mut self,
+        thread: remoteprocess::Thread,
+        lock: remoteprocess::Lock,
+    ) -> Result<Vec<StackFrame>> {
+        let mut native_ips: Vec<u64> = Vec::new();
+        for ip in self.unwinder.cursor(&thread)? {
+            let ip = ip?;
+            native_ips.push(ip)
+        }
+
+        drop(lock);
+
+        let mut native_frames: Vec<StackFrame> = Vec::new();
+        for ip in native_ips {
+            self.symbolicate(ip, &mut |sf| {
+                let name: &str = sf
+                    .function
+                    .as_ref()
+                    .map(|s| s as &str)
+                    .unwrap_or_else(|| "cfunc (unnamed)".into());
```

Comment:
If there's a chance that this will be printed, it should probably use the "[c function]" notation like the existing C func code does.